server/controllers/additional-sheet-parser.js

```diff
@@ -0,0 +1,89 @@
+// Parses "optimization_paramters," expression data sheets, and 2-column sheets
+// from GRNmap input or output workbook
+
+var TWO_COL_SHEET_NAMES = [
+    "production_rates",
+    "degradation_rates",
+    "threshold_b",
+    "optimized_production_rates",
+    "optimized_threshold_b"];
+
+var EXPRESSION_SHEET_SUFFIXES = ["_expression", "_optimized_expression", "_sigmas"];
+
+var isExpressionSheet = function (sheetName) {
+    for (var i = 0; i < EXPRESSION_SHEET_SUFFIXES.length; i++) {
+        var suffix = EXPRESSION_SHEET_SUFFIXES[i];
+        if (sheetName.includes(suffix)) {
+            return true;
+        }
+    }
+    return false;
+};
+
+// Helper function, to be replaced with the includes() method when GRNsight is ported to ES6
+var isInArray = function (value, array) {
+    return array.indexOf(value) > -1;
+};
+
+var fillArray = function (value, array, length) { // mutator
+    while (array.length < length) {
+        array.push(value);
+    }
+    return array;
+};
+
+module.exports = function (workbook) {
+
+    var output = {};
+    output["expression"] = {}; // expression data
+    output["meta"] = {};
+    // TODO: Need better name to generalize 2-column data
+    output["test"] = {}; // 2-column data
```

Comment:
Why isn't this just:
```javascript
var output = {
    expression: {},
    meta: {},
    test: {}
};
```
.....I think this conveys the structure a little better.